scanner/decisions/0010-nvd-cvss-scores.md

```diff
@@ -0,0 +1,192 @@
+        string vector = 2;
+      }
+      message V3 {
+        float base_score = 1;
+        string vector = 2;
+      }
+      V2 v2 = 1;
+      V3 v3 = 2;
+    }
+    ...
+    string severity = 6;
+    ...
+    CVSS cvss = 12;
+  }
+  ...
+}
+```
+
+## Decision
+
+The VulnerabilityReport API will be extended as follows:
+
+```
+message VulnerabilityReport {
+  message Vulnerability {
+    enum Severity {
+      SEVERITY_UNSPECIFIED = 0;
+      SEVERITY_LOW = 1;
+      SEVERITY_MODERATE = 2;
+      SEVERITY_IMPORTANT = 3;
+      SEVERITY_CRITICAL = 4;
+    }
+    message CVSS {
+      enum Source { <-- New field
+        SOURCE_UNKNOWN = 0;
+        SOURCE_RED_HAT = 1;
+        SOURCE_OSV = 2;
+        SOURCE_NVD = 3;
+      }
+      message V2 {
+        float base_score = 1;
+        string vector = 2;
+      }
+      message V3 {
+        float base_score = 1;
+        string vector = 2;
+      }
+      V2 v2 = 1;
+      V3 v3 = 2;
+      Source source = 3; <-- New field
+      string url = 4;    <-- New field
+    }
+    ...
+    string severity = 6;
+    ...
+    CVSS cvss = 12;
+    repeated CVSS cvss_metrics = 13; <-- New field.
+  }
+  ...
+}
+```
+
+There will be a new type plus two new fields added:
+
+* **enum** `CVSS.Source`
+  * This is the **enum** that specifies the supported CVSS metrics data sources.
+* **field** `CVSS.source`
+  * This **field** specifies the source of the particular CVSS metrics.
+* `CVSS.url`
+  * This specifies the CVSS score source URL, aiding both API and UI users in tracking the origin of this metric.
+* `cvss_metrics`
+  * This is a list of each unique CVSS metric based on the source.
+
+The original `cvss` field will remain and will continue to represent the Scanner's preferred CVSS score.
+This is currently the score from the vulnerability's original data source, if available, otherwise NVD.
+
+**JSON example**:
+
+
+```sh
+"9216828": {
+	"id": "9216828",
+	"name": "CVE-2023-48231",
+	"description": "DOCUMENTATION: A heap use-after-free flaw was found in the vim package. ... MITIGATION: Mitigation for this issue is either not available or ..., applicability to widespread installation base or stability.",
+	"issued": {
+		"seconds": -62135596800
+	},
+	"link": "https://access.redhat.com/security/cve/CVE-2023-48231",
+	"severity": "Low",
+	"normalized_severity": 1,
+	"cvss": {
+		"source": "SOURCE_RED_HAT",
+		"v3": {
+			"base_score": 4.3,
+			"vector": "CVSS:3.1/AV:N/AC:L/PR:N/UI:R/S:U/C:N/I:N/A:L"
+			},
+		"url":"https://access.redhat.com/security/cve/CVE-2023-48231"
+		},
+	"cvss_metrics": [{
+		"source": "SOURCE_RED_HAT",
+		"v3": {
+			"base_score": 4.3,
+			"vector": "CVSS:3.1/AV:N/AC:L/PR:N/UI:R/S:U/C:N/I:N/A:L"
+			},
+		"url":"https://access.redhat.com/security/cve/CVE-2023-48231"
+		},{
+		"source": "SOURCE_NVD",
+		"v3": {
+			"base_score": 4.3,
+			"vector": "CVSS:3.1/AV:N/AC:L/PR:N/UI:R/S:U/C:N/I:N/A:L"
+			},
+		"url":"https://nvd.nist.gov/vuln/detail/CVE-2023-48231"
+		}]
+}
+```
```

Comment:
```suggestion
```json
"9216828": {
	"id": "9216828",
	"name": "CVE-2023-48231",
	"description": "DOCUMENTATION: A heap use-after-free flaw was found in the vim package. ... MITIGATION: Mitigation for this issue is either not available or ..., applicability to widespread installation base or stability.",
	"issued": {
		"seconds": -62135596800
	},
	"link": "https://access.redhat.com/security/cve/CVE-2023-48231",
	"severity": "Low",
	"normalized_severity": 1,
	"cvss": {
		"source": "SOURCE_RED_HAT",
		"v3": {
			"base_score": 4.3,
			"vector": "CVSS:3.1/AV:N/AC:L/PR:N/UI:R/S:U/C:N/I:N/A:L"
		},
		"url": "https://access.redhat.com/security/cve/CVE-2023-48231"
	},
	"cvss_metrics": [
		{
			"source": "SOURCE_RED_HAT",
			"v3": {
				"base_score": 4.3,
				"vector": "CVSS:3.1/AV:N/AC:L/PR:N/UI:R/S:U/C:N/I:N/A:L"
			},
			"url": "https://access.redhat.com/security/cve/CVE-2023-48231"
		}, {
			"source": "SOURCE_NVD",
			"v3": {
				"base_score": 4.3,
				"vector": "CVSS:3.1/AV:N/AC:L/PR:N/UI:R/S:U/C:N/I:N/A:L"
			},
			"url": "https://nvd.nist.gov/vuln/detail/CVE-2023-48231"
		}
	]
}
```
```
nit: formatted json and added json syntax coloring to make it easier to read

Reply:
Agreed with this ^. Can you commit this suggestion (though, make sure nothing has unexpectedly changed)?